controllers/admin/base_admin_controller.js

```diff
@@ -0,0 +1,75 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module.exports = function BaseAdminControllerModule(pb) {
+  "use strict";
+
+  var util = pb.util;
+  var BaseController = pb.BaseController;
+
+  /**
+   * This class serves as a base class for all the controllers used in the admin control panel
+   * @constructor
+   */
+  function BaseAdminController() {}
```

Comment:
We need to add the following to ensure that the prototype shows up correctly in the auto generated documentation:
```
@class BaseAdminController
@constructor
@extends BaseController
```